converter/src/main/java/gov/cms/qpp/conversion/validate/Checker.java

```diff
@@ -134,34 +133,18 @@ public Checker singleValue(LocalizedError code, String name) {
 	 * @return The checker, for chaining method calls
 	 */
 	public Checker isValidDate(LocalizedError code, String name) {
-		String date = node.getValue(name);
-		try {
-			String parse = cleanString(date);
-			parse = parse.replace("-", "").replace("/", "");
-			if (parse.length() > DATE_FORMAT.length()) {
-				parse = parse.substring(0, DATE_FORMAT.length());
+		lastAppraised = node.getValue(name);
+		if (!shouldShortcut()) {
+			try {
+				LocalDate.parse(node.getValue(name), DateTimeFormatter.ofPattern(DATE_FORMAT));
```

Comment:
This code now fails with an ISO8061 date/time.  Since `DATE_FORMAT` is `yyyyMMdd`, this fails for anything that isn't exactly in that format.  Can we tell `LocalDate` to be agnostic to the date format but can still parse out a year?  Or perhaps we can use `Instant`?

Reply:
@halprin I was looking into Optional sections for LocalDateTime but it wasn't working for me. I tried Instant, but was still restricted. For now, I changed the code back to cleaning the string and parsing it and simply created a new class to house all the code in one place.